In [31]:
from IPython.display import IFrame

In [32]:
IFrame('https://upload.wikimedia.org/wikipedia/commons/thumb/e/e1/Logo_of_YouTube_%282015-2017%29.svg/640px-Logo_of_YouTube_%282015-2017%29.svg.png', width=700, height=300)

In [29]:
import pandas as pd

# Loading different countries data

In [2]:
df = pd.read_csv(r"C:\Users\user\Desktop\Elevate Labs Internship\Projects\Project 2\youtube.csv")

In [4]:
df.head(5)

,index,video_id,trending_date,title,channel_title,category_id,publish_date,time_frame,published_day_of_week,publish_country,tags,views,likes,dislikes,comment_count,comments_disabled,ratings_disabled,video_error_or_removed
0,0,2kyS6SvSYSE,17.14.11,WE WANT TO TALK ABOUT OUR MARRIAGE,CaseyNeistat,22,13/11/2017,17:00 to 17:59,Monday,US,SHANtell martin,748374,57527,2966,15954,False,False,False
1,1,1ZAPwfrtAFY,17.14.11,The Trump Presidency: Last Week Tonight with J...,LastWeekTonight,24,13/11/2017,7:00 to 7:59,Monday,US,"last week tonight trump presidency|""last week ...",2418783,97185,6146,12703,False,False,False
2,2,5qpjK5DgCt4,17.14.11,"Racist Superman | Rudy Mancuso, King Bach & Le...",Rudy Mancuso,23,12/11/2017,19:00 to 19:59,Sunday,US,"racist superman|""rudy""""|""""mancuso""""|""""king""""|""...",3191434,146033,5339,8181,False,False,False
3,3,puqaWrEC7tY,17.14.11,Nickelback Lyrics: Real or Fake?,Good Mythical Morning,24,13/11/2017,11:00 to 11:59,Monday,US,"rhett and link|""gmm""""|""""good mythical morning""...",343168,10172,666,2146,False,False,False
4,4,d380meD0W0M,17.14.11,I Dare You: GOING BALD!?,nigahiga,24,12/11/2017,18:00 to 18:59,Sunday,US,"ryan|""higa""""|""""higatv""""|""""nigahiga""""|""""i dare ...",2095731,132235,1989,17518,False,False,False


# Data Analysis

In [7]:
print(f"Rows {df.shape[0]}, columns {df.shape[1]}.")

Rows 161470, columns 18.


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 161470 entries, 0 to 161469
Data columns (total 18 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   index                   161470 non-null  int64 
 1   video_id                161470 non-null  object
 2   trending_date           161470 non-null  object
 3   title                   161470 non-null  object
 4   channel_title           161470 non-null  object
 5   category_id             161470 non-null  int64 
 6   publish_date            161470 non-null  object
 7   time_frame              161470 non-null  object
 8   published_day_of_week   161470 non-null  object
 9   publish_country         161470 non-null  object
 10  tags                    161470 non-null  object
 11  views                   161470 non-null  int64 
 12  likes                   161470 non-null  int64 
 13  dislikes                161470 non-null  int64 
 14  comment_count           161470 non-n

In [8]:
df.isnull().sum()

index                     0
video_id                  0
trending_date             0
title                     0
channel_title             0
category_id               0
publish_date              0
time_frame                0
published_day_of_week     0
publish_country           0
tags                      0
views                     0
likes                     0
dislikes                  0
comment_count             0
comments_disabled         0
ratings_disabled          0
video_error_or_removed    0
dtype: int64

# There are 161470 rows and 18 columns in the dataset.
# There are no null values in in the dataset.
# The Datatype of `trending_date`, `publish_date` and `time_frame` are object, that will be changed into correct date format.

# Datatype of `trending_date` and `publish_date` will be Standardized

In [37]:
print(df[['trending_date', 'publish_date']].head())

  trending_date publish_date
0    2017-11-14   2017-11-13
1    2017-11-14   2017-11-13
2    2017-11-14   2017-11-12
3    2017-11-14   2017-11-13
4    2017-11-14   2017-11-12


In [46]:
df['trending_date'] = pd.to_datetime(df['trending_date'], format='%Y-%m-%d')
df['publish_date'] = pd.to_datetime(df['publish_date'], format='%Y-%m-%d')

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 161470 entries, 0 to 161469
Data columns (total 22 columns):
 #   Column                  Non-Null Count   Dtype         
---  ------                  --------------   -----         
 0   index                   161470 non-null  int64         
 1   video_id                161470 non-null  object        
 2   trending_date           161470 non-null  datetime64[ns]
 3   title                   161470 non-null  object        
 4   channel_title           161470 non-null  object        
 5   category_id             161470 non-null  int64         
 6   publish_date            161470 non-null  datetime64[ns]
 7   time_frame              161470 non-null  object        
 8   published_day_of_week   161470 non-null  object        
 9   publish_country         161470 non-null  object        
 10  tags                    161470 non-null  object        
 11  views                   161470 non-null  int64         
 12  likes                   161470

In [48]:
df.head()

,index,video_id,trending_date,title,channel_title,category_id,publish_date,time_frame,published_day_of_week,publish_country,...,likes,dislikes,comment_count,comments_disabled,ratings_disabled,video_error_or_removed,title_sentiment,tags_sentiment,title_sentiment_label,tags_sentiment_label
0,0,2kyS6SvSYSE,2017-11-14,WE WANT TO TALK ABOUT OUR MARRIAGE,CaseyNeistat,22,2017-11-13,17:00 to 17:59,Monday,US,...,57527,2966,15954,0,0,0,0.0772,0.0000,Positive,Neutral
1,1,1ZAPwfrtAFY,2017-11-14,The Trump Presidency: Last Week Tonight with J...,LastWeekTonight,24,2017-11-13,7:00 to 7:59,Monday,US,...,97185,6146,12703,0,0,0,0.0000,0.0000,Neutral,Neutral
2,2,5qpjK5DgCt4,2017-11-14,"Racist Superman | Rudy Mancuso, King Bach & Le...",Rudy Mancuso,23,2017-11-12,19:00 to 19:59,Sunday,US,...,146033,5339,8181,0,0,0,-0.6124,-0.6124,Negative,Negative
3,3,puqaWrEC7tY,2017-11-14,Nickelback Lyrics: Real or Fake?,Good Mythical Morning,24,2017-11-13,11:00 to 11:59,Monday,US,...,10172,666,2146,0,0,0,-0.4767,0.4404,Negative,Positive
4,4,d380meD0W0M,2017-11-14,I Dare You: GOING BALD!?,nigahiga,24,2017-11-12,18:00 to 18:59,Sunday,US,...,132235,1989,17518,0,0,0,0.0000,0.0000,Neutral,Neutral


# Converting boolean columns to 0/1, so that MySQL can accept them as integers

In [38]:
bool_cols = ['comments_disabled', 'ratings_disabled', 'video_error_or_removed']
for col in bool_cols:
    df[col] = df[col].map({False: 0, True: 1, 'False': 0, 'True': 1})

In [39]:
df.head(5)

,index,video_id,trending_date,title,channel_title,category_id,publish_date,time_frame,published_day_of_week,publish_country,...,likes,dislikes,comment_count,comments_disabled,ratings_disabled,video_error_or_removed,title_sentiment,tags_sentiment,title_sentiment_label,tags_sentiment_label
0,0,2kyS6SvSYSE,2017-11-14,WE WANT TO TALK ABOUT OUR MARRIAGE,CaseyNeistat,22,2017-11-13,17:00 to 17:59,Monday,US,...,57527,2966,15954,0,0,0,0.0772,0.0000,Positive,Neutral
1,1,1ZAPwfrtAFY,2017-11-14,The Trump Presidency: Last Week Tonight with J...,LastWeekTonight,24,2017-11-13,7:00 to 7:59,Monday,US,...,97185,6146,12703,0,0,0,0.0000,0.0000,Neutral,Neutral
2,2,5qpjK5DgCt4,2017-11-14,"Racist Superman | Rudy Mancuso, King Bach & Le...",Rudy Mancuso,23,2017-11-12,19:00 to 19:59,Sunday,US,...,146033,5339,8181,0,0,0,-0.6124,-0.6124,Negative,Negative
3,3,puqaWrEC7tY,2017-11-14,Nickelback Lyrics: Real or Fake?,Good Mythical Morning,24,2017-11-13,11:00 to 11:59,Monday,US,...,10172,666,2146,0,0,0,-0.4767,0.4404,Negative,Positive
4,4,d380meD0W0M,2017-11-14,I Dare You: GOING BALD!?,nigahiga,24,2017-11-12,18:00 to 18:59,Sunday,US,...,132235,1989,17518,0,0,0,0.0000,0.0000,Neutral,Neutral


In [19]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

# Creating sentiment and label columns for tags & title

In [21]:
analyzer = SentimentIntensityAnalyzer()
df['title_sentiment'] = df['title'].apply(lambda x: analyzer.polarity_scores(x)['compound'])
df['tags_sentiment'] = df['tags'].apply(lambda x: analyzer.polarity_scores(x)['compound'])

In [22]:
def sentiment_label(x):
    if x >= 0.05:
        return 'Positive'
    elif x <= -0.05:
        return 'Negative'
    else:
        return 'Neutral'

df['title_sentiment_label'] = df['title_sentiment'].apply(sentiment_label)
df['tags_sentiment_label'] = df['tags_sentiment'].apply(sentiment_label)

In [23]:
df.head()

,index,video_id,trending_date,title,channel_title,category_id,publish_date,time_frame,published_day_of_week,publish_country,...,likes,dislikes,comment_count,comments_disabled,ratings_disabled,video_error_or_removed,title_sentiment,tags_sentiment,title_sentiment_label,tags_sentiment_label
0,0,2kyS6SvSYSE,2017-11-14,WE WANT TO TALK ABOUT OUR MARRIAGE,CaseyNeistat,22,2017-11-13,17:00 to 17:59,Monday,US,...,57527,2966,15954,False,False,False,0.0772,0.0000,Positive,Neutral
1,1,1ZAPwfrtAFY,2017-11-14,The Trump Presidency: Last Week Tonight with J...,LastWeekTonight,24,2017-11-13,7:00 to 7:59,Monday,US,...,97185,6146,12703,False,False,False,0.0000,0.0000,Neutral,Neutral
2,2,5qpjK5DgCt4,2017-11-14,"Racist Superman | Rudy Mancuso, King Bach & Le...",Rudy Mancuso,23,2017-11-12,19:00 to 19:59,Sunday,US,...,146033,5339,8181,False,False,False,-0.6124,-0.6124,Negative,Negative
3,3,puqaWrEC7tY,2017-11-14,Nickelback Lyrics: Real or Fake?,Good Mythical Morning,24,2017-11-13,11:00 to 11:59,Monday,US,...,10172,666,2146,False,False,False,-0.4767,0.4404,Negative,Positive
4,4,d380meD0W0M,2017-11-14,I Dare You: GOING BALD!?,nigahiga,24,2017-11-12,18:00 to 18:59,Sunday,US,...,132235,1989,17518,False,False,False,0.0000,0.0000,Neutral,Neutral


# Performing Sentiment Analysis on Titles & Tags

In [24]:
df['tags_sentiment_label'].value_counts()

tags_sentiment_label
Neutral     122829
Positive     23730
Negative     14911
Name: count, dtype: int64

In [25]:
df['title_sentiment_label'].value_counts()

title_sentiment_label
Neutral     100168
Positive     33892
Negative     27410
Name: count, dtype: int64

In [26]:
df[['title', 'title_sentiment', 'title_sentiment_label']].sort_values(by='title_sentiment', ascending=False).head(10)

,title,title_sentiment,title_sentiment_label
91718,Guitar Music Cover - Greatest Love Songs Of Al...,0.9808,Positive
132421,Guitar Music Cover - Greatest Love Songs Of Al...,0.9808,Positive
91923,Guitar Music Cover - Greatest Love Songs Of Al...,0.9808,Positive
132619,Guitar Music Cover - Greatest Love Songs Of Al...,0.9808,Positive
98136,Relaxing Instrumental Music 2018 - Be worthy l...,0.9623,Positive
156853,Best Guitar Relaxing Instrumental Music - Top ...,0.9618,Positive
115681,Best Guitar Relaxing Instrumental Music - Top ...,0.9618,Positive
115945,Best Guitar Relaxing Instrumental Music - Top ...,0.9618,Positive
156537,Best Guitar Relaxing Instrumental Music - Top ...,0.9618,Positive
136910,Top 10 Best Super Bowl Commercials 2018 (Best ...,0.9584,Positive


In [27]:
df[['tags', 'tags_sentiment', 'tags_sentiment_label']].sort_values(by='tags_sentiment', ascending=False).head(10)

,tags,tags_sentiment,tags_sentiment_label
97818,"Overwatch Funny Moments Best Plays Montage""|""O...",0.9973,Positive
120849,"gift cards|""free amazon gift cards""|""free gift...",0.9968,Positive
120997,"gift card|""free google play gift card codes""|""...",0.9962,Positive
120677,"gift card|""free amazon gift card""|""amazon gift...",0.9959,Positive
112513,"beyblade burst super zetsu|""beyblade burst sup...",0.9939,Positive
91956,"Dragon Ball Super 124 Vostfr Preview""|""theloga...",0.9939,Positive
155152,"best future talents moments auditions|""global ...",0.9936,Positive
90868,"preview""|""vostfr""|""dragon""|""ball""|""super""|""the...",0.9927,Positive
91088,"preview""|""vostfr""|""dragon""|""ball""|""super""|""the...",0.9927,Positive
88209,Dragon Ball Super Episode 122 English Subbed P...,0.9921,Positive


# Converting to CSV for MySQL Analysis

In [52]:
df.to_csv(r"C:\Users\user\Desktop\Elevate Labs Internship\Projects\Project 2\youtube_trend.csv", index = False, encoding = "utf-8-sig")